question26/js/login.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const submitButton = document.getElementById('js-submit-button');
+const userInfoInputArea = document.getElementById('name_mail');
+const passwordInputArea = document.getElementById('password');
+const form = document.getElementById('form');
+
+if(localStorage.hasOwnProperty('token')) {
+  window.location.href = './newsComponents/index.html'
+}
+
+const isValidStatus = {
+  name_mail: false,
+  password: false,
+}
+
+const validationInfo = {
+  name_mail: {
+    validation: (value) => value
+  },
+  password: {
+    validation : (value) => /(?=.*[a-zA-Z])(?=.*[0-9])[a-zA-z0-9]{8,}/.test(value),
+    errorMessage: '※8文字以上の大小の英数字を交ぜたものにしてください。'  
+  }
+}
+
+const setValidationEvents = (isValid, targetForm) => {
+  if (isValid) {
+    targetForm.classList.add('valid');
+    targetForm.classList.remove('invalid');
+    targetForm.nextElementSibling.textContent = ''; 
+    isValidStatus[targetForm.id] = true;
+  } else {
+    targetForm.classList.add('invalid');
+    targetForm.classList.remove('valid');
+    targetForm.nextElementSibling.textContent = validationInfo[targetForm.id].errorMessage;
+    isValidStatus[targetForm.id] = false; 
+  }
+}
+
+const validateInputValue = (e) => {
+  const targetForm = e.target;
+  const value = targetForm.value.trim();
+  const result = validationInfo[targetForm.id].validation(value);
+  setValidationEvents(result,targetForm);
+  renderRequiredFieldMessages(targetForm);  
+  switchSubmitButton(checkAllValidity());
+} 
+
+const checkAllValidity = () => {
+  return Object.values(isValidStatus).every((result) => result );
+}
+
+const switchSubmitButton = (isValid) => {
+  submitButton.disabled = isValid ? false : true;
+} 
+
+const renderRequiredFieldMessages = (targetForm) => {
+  if (targetForm.value.trim() === "") {
+    targetForm.nextElementSibling.textContent = "※入力必須項目です";
+  }
+};
+
+userInfoInputArea.addEventListener('blur', validateInputValue);
+passwordInputArea.addEventListener('blur', validateInputValue);
+
+
+const checkData = (inputData) => {
```

Comment:
IMO
`Destructuring assignment` can be used
```suggestion
const checkData = ({name_mail,password}) => {
```

Reply:
Thank you for pointing it out! I had not thought of that idea. That would make it more readable. Fixed so please confirm.

fixed commit : https://github.com/csakyo/homework/pull/40/commits/af913642abc18c3ef82989848b98b26014fa8611

Reply:
Q. `name_mail`,` user_password`
How do you use snake_case and camelCase?

Reply:
When I write JS, I try to be consistent with CamelCase. However, in this case, when I use formData to retrieve data, it automatically retrieves the value of the HTML name attribute, and that is where the difference occurs.

Does that answer your question...?

![スクリーンショット 2022-10-08 15 53 50](https://user-images.githubusercontent.com/51282804/194695972-e6a362e1-795a-4a60-a3de-1da23d782af7.png)

![スクリーンショット 2022-10-08 16 28 43](https://user-images.githubusercontent.com/51282804/194695744-96440a4f-0cd5-47e4-9c7b-6561624fd8bc.png)

Reply:
Thank you.
Understood :)